skimage/morphology/max_tree.py

```diff
@@ -0,0 +1,667 @@
+           A. Giuliani (Eds.), Medical Data Analysis. Lecture Notes in Computer
+           Science, vol 2526, pp. 210-220. Springer Berlin Heidelberg.
+           DOI:10.1007/3-540-36104-9_23
+    .. [2] Carlinet, E., & Geraud, T. (2014). A Comparative Review of
+           Component Tree Computation Algorithms. IEEE Transactions on Image
+           Processing, 23(9), 3885-3895.
+           DOI:10.1109/TIP.2014.2336551
+
+    Examples
+    --------
+    We create an image (quadratic function with a maximum in the center and
+    4 additional local maxima.
+
+    >>> w = 12
+    >>> x, y = np.mgrid[0:w,0:w]
+    >>> f = 20 - 0.2*((x - w/2)**2 + (y-w/2)**2)
+    >>> f[2:3,1:5] = 40; f[2:4,9:11] = 60; f[9:11,2:4] = 80
+    >>> f[9:10,9:11] = 100; f[10,10] = 100
+    >>> f = f.astype(np.int)
+
+    We can calculate the diameter opening:
+
+    >>> open = diameter_opening(f, 3, connectivity=1)
+
+    The peaks with a maximal extension of 2 or less are removed.
+    The remaining peaks have all a maximal extension of at least 3.
+    """
+    output = image.copy()
+
+    if parent is None or tree_traverser is None:
+        parent, tree_traverser = max_tree(image, connectivity)
+
+    diam = _max_tree._compute_extension(image.ravel(),
+                                        np.array(image.shape, dtype=np.int32),
+                                        parent.ravel(), tree_traverser)
+
+    _max_tree._direct_filter(image.ravel(), output.ravel(), parent.ravel(),
+                             tree_traverser, diam, diameter_threshold)
+    return output
+
+
+def area_closing(image, area_threshold=25, connectivity=2,
+                 parent=None, tree_traverser=None):
+    """Performs an area closing of the image.
+
+    Area closing removes all dark structures of an image with
+    a surface smaller than area_threshold.
+    The output image is larger than or equal to the input image
+    for every pixel and all local minima have at least a surface of
+    area_threshold pixels.
+
+    Area closings are similar to morphological closings, but
+    they do not use a fixed structuring element, but rather a deformable
+    one, with surface = area_threshold.
+
+    In the binary case, area closings are equivalent to
+    remove_small_holes; this operator is thus extended to grey-level images.
+
+    Technically, this operator is based on the max-tree representation of
+    the image.
+
+    Parameters
+    ----------
+    image: ndarray
+        The input image for which the area_closing is to be calculated.
+        This image can be of any type.
+    area_threshold: unsigned int
+        The size parameter (number of pixels). The default value is arbitrary.
+    connectivity: unsigned int, optional
+        The neighborhood connectivity. The integer represents the maximum
+        number of orthogonal steps to reach a neighbor. In 2D, it is 1 for
+        4-connectivity and 2 for 8-connectivity. Default value is 1.
```

Comment:
This default value does not match the default value in the signature... ;)

Reply:
Yep, true. 
I set everything to 1.